### PR TITLE
[APPSEC-119] Modify rule S6287: Educational content (Java)

### PR DESCRIPTION
**Pull request by pedro-oliveira-sonarsource**
Specification ticket: [APPSEC-119](https://sonarsource.atlassian.net/browse/APPSEC-119).

### CODE REVIEW

File: rules/S6287/java/how-to-fix-it/java-se.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                                            .get());
+    if (cookieOpt.isPresent() ) {
+        response.sendRedirect("/welcome.jsp");
+    } else {
+        String cookie = request.getParameter("cookie");
+        Cookie cookieObj = new Cookie("jsessionid", cookie);  // Noncompliant
+        response.addCookie(cookieObj);
+
+        response.sendRedirect("/welcome.jsp");
+    }
```
Comment (loris-s-sonarsource) [inline rules/S6287/java/how-to-fix-it/java-se.adoc:27]: I found a code turn more concise:

```suggestion
    Optional<Cookie> cookieOpt = Arrays.stream(request.getCookies())
                                    .filter(c -> c.getName().equals("jsessionid"))
                                    .findFirst();

      if (!cookieOpt.isPresent()) {
          String cookie = request.getParameter("cookie");
          Cookie cookieObj = new Cookie("jsessionid", cookie);  // Noncompliant
          response.addCookie(cookieObj);
      }

      response.sendRedirect("/welcome.jsp");
```

wdyt?

File: rules/S6287/java/how-to-fix-it/java-se.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                                            .filter(t -> "jsessionid"
+                                            .equals(t.getName()))
+                                            .findFirst()
+                                            .get());
+
+    if (cookieOpt.isPresent()) {
+        response.sendRedirect("/welcome.jsp");
+    } else {
+        response.sendRedirect("/cookie.jsp");
+    }
```
Comment (loris-s-sonarsource) [inline rules/S6287/java/how-to-fix-it/java-se.adoc:47]:
```suggestion
    Optional<Cookie> cookieOpt = Arrays.stream(request.getCookies())
                                    .filter(c -> c.getName().equals("jsessionid"))
                                    .findFirst();

      if (!cookieOpt.isPresent()) {
          response.sendRedirect("/getCookie.jsp");
      } else {
          response.sendRedirect("/welcome.jsp");
      }
```

**Review by loris-s-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S6287/java/how-to-fix-it/java-se.adoc (hunk 1 above)
Comment (pedro-oliveira-sonarsource) [reply rules/S6287/java/how-to-fix-it/java-se.adoc:27]: Looks cleaner.

File: rules/S6287/java/how-to-fix-it/java-se.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                                    .filter(c -> c.getName().equals("jsessionid"))
+                                    .findFirst();
+
+      if (!cookieOpt.isPresent()) {
+          String cookie = request.getParameter("cookie");
+          Cookie cookieObj = new Cookie("jsessionid", cookie);  // Noncompliant
+          response.addCookie(cookieObj);
+      }
+
+      response.sendRedirect("/welcome.jsp");
```
Comment (loris-s-sonarsource) [inline rules/S6287/java/how-to-fix-it/java-se.adoc:22]:
```suggestion
    Optional<Cookie> cookieOpt = Arrays.stream(request.getCookies())
                                    .filter(c -> c.getName().equals("jsessionid"))
                                    .findFirst();

    if (!cookieOpt.isPresent()) {
        String cookie = request.getParameter("cookie");
        Cookie cookieObj = new Cookie("jsessionid", cookie);  // Noncompliant
        response.addCookie(cookieObj);
    }

    response.sendRedirect("/welcome.jsp");
```

File: rules/S6287/java/how-to-fix-it/java-se.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Optional<Cookie> cookieOpt = Arrays.stream(request.getCookies())
+                                    .filter(c -> c.getName().equals("jsessionid"))
+                                    .findFirst();
+
+      if (!cookieOpt.isPresent()) {
+          response.sendRedirect("/getCookie.jsp");
+      } else {
+          response.sendRedirect("/welcome.jsp");
+      }
+}
```
Comment (loris-s-sonarsource) [inline rules/S6287/java/how-to-fix-it/java-se.adoc:39]:
```suggestion
protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
    Optional<Cookie> cookieOpt = Arrays.stream(request.getCookies())
                                    .filter(c -> c.getName().equals("jsessionid"))
                                    .findFirst();

    if (!cookieOpt.isPresent()) {
        response.sendRedirect("/getCookie.jsp");
    } else {
        response.sendRedirect("/welcome.jsp");
    }
}
```

**Review by loris-s-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**